codespell_lib/tests/test_basic.py

```diff
@@ -393,6 +393,41 @@ def test_case_handling(tmpdir, capsys):
         assert f.read().decode('utf-8') == 'this has an ASCII error'
 
 
+def test_case_handling_in_fixes(tmpdir, capsys):
+    """Test that the case of fixes is similar to the mispelled word."""
+    d = str(tmpdir)
+    with open(op.join(d, 'dictionary_with_reason.txt'), 'w') as f:
+        fix = 'adopter, adaptor'
+        reason = 'these are different words!'
+        f.write(f'adoptor->{fix}, {reason}\n')
+    dictionary_name = f.name
```

Comment:
Is this really useful? Good code coverage in tests should be enough, and I think we do have good coverage.

Reply:
I'd say yes, the no-fix format is:
`adoptor->adopter, adaptor,`

So we need to check it handles the trailing comma and doesn't put one on the offered suggestions for example (which a naive split/upper/join might). Plus the non-reason format is what we're actually using in our dictionaries.

Reply:
Precisely, isn't that already checked elsewhere? In any case, such checks seem orthogonal to this PR. I mean, I don't mind adding additional checks in another PR, but I feel this PR is "hijkacked" to ask for more tests than are directly related to this PR.

Reply:
> Precisely, isn't that already checked elsewhere? In any case, such checks seem orthogonal to this PR. I mean, I don't mind adding additional checks in another PR

I'd disagree, when this is merged, the code will be:
```
def fix_case(word, fixword):
    if word == word.capitalize():
        return ', '.join(w.strip().capitalize() for w in fixword.split(','))
    elif word == word.upper():
        return fixword.upper()
    # they are both lower case
    # or we don't have any idea
    return fixword
```

By observation, if fixword was passed in with the trailing comma, then capital case would either strip it off, or return an extra space on the end of the line, but upper and unmodified would pass it back out unmodified. i.e. the behaviour varies between the modes, which I'd say means we should test it while we're touching this function.

I've checked the other code and it does currently strip the trailing comma with no reason, but we've got no guarantee of that in this code.

> "hijkacked"

Can you add `hijkacked->hijacked` ? 😆

Reply:
I see. I do know that this function does not get trailing commas currently, but I understand there's no guarantee this won't happen in the future.

I can add the test, but it would be better spending time on changing `fixword` to be a real list before it gets to this function. But then we would need to make sure the code changing the comma-separated words from the dictionary into a list does the right thing. However, I feel that would be better handled in unit tests.

Reply:
How does it look like now?